javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,169 @@
+    if (typeof command !== 'string') {
+        return 'Please give command in correct format';
+    }
+    //convert command into an array
+    const commandArray = command.split(' ');
+    //if the first word of comman is hello then reply with wellcome message
+    if (commandArray[0].toLowerCase() === 'hello') {
+        if (name === '') {
+            name = commandArray[commandArray.length - 1]
+            return 'Nice to meet you ' + name;
+        }
+        else {
+            return 'Why are you asking again?';
+        }
+    }
+    //if the first word is 'what' then further check the last word to perform appropriate operation
+    else if (commandArray[0].toLowerCase() === 'what') {
+        if (commandArray[commandArray.length - 1].toLowerCase().includes('name')) {
+            if (name !== '') {
+                return 'Your name is ' + name;
+            }
+            else {
+
+                return 'Please Enter your name first';
+            }
+
+        }
+        else if (commandArray[commandArray.length - 1].toLowerCase().includes('todo')) {
+            //show todo list
+            return getToDos(todos)
+        }
+        else if (commandArray[commandArray.length - 1].toLowerCase().includes('today')) {
+            //show today date
+            return getDate();
+        }
+        else if (commandArray[commandArray.length - 1].toLowerCase().includes('address')) {
+            //show address
+            return "hackyourfuture's address is: Enghavevej 80 2450 københavn sv";
+        }
+
+        else {
+            //call function to perform math opration
+            return getMathResult(commandArray);
+        }
+
+    }
+    else if (commandArray[0].toLowerCase() === 'add') {
+        //call function to add todo in todos list
+        return addTodo(commandArray);
+    }
+    else if (commandArray[0].toLowerCase() === 'remove') {
+        //call function to remove todo from todos list
+        return removeTodo(commandArray);
+    }
+    else if (commandArray[0].toLowerCase() === 'set') {
+        //call function to set timer
+        return setTimer(commandArray);
+    }
+    else {
+        return "Oops I don't recognize your command please follow program manual";
+    }
+    //end main program
+}
+
+/**return the todos list*/
+function getToDos(todos) {
+    let todo = '';
+    for (let i = 0; i < todos.length - 1; i++) {
+        if (todos.length === 2) {
+            todo += todos[i]
+        }
+        else {
+            todo += todos[i] + ',';
+        }
+    }
+    if (todos.length > 1) {
+        todo += ' and ' + todos[todos.length - 1];
+    }
+    else if (todos.length === 1) {
+        todo = todos[0]
+    }
+    return 'You have ' + todos.length + ' todos ' + todo;
+}
+/**perform math opration and return result */
+function getMathResult(commandArray) {
+    const operator = commandArray[3]
+    const a = parseInt(commandArray[2]);
+    const b = parseInt(commandArray[4]);
```

Comment:
You can try to use meaningful variable names instead of a and b

Reply:
done